From 7ad611486e5bc890410937939465882af616e9e0 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 24 Aug 2022 06:11:47 +0300
Subject: [PATCH] [SW-86869] gaudi3: set const to 0 in mem init function

Since the function arcboot_rom_mem_init which uses mem copy block
to initialize memory can be used outside ROM, this patch sets
const register to zero to initialize mem with all zeros.

Change-Id: Ic61510b7b69605277dde9ee52f3203245a009d3e
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index f3e4309..cfaaae5 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -68,6 +68,7 @@
 	mem_cfg |= D0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_M;
 
 	regmap_writel(mem_cfg, mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2);
+	regmap_writel(0x0, mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CONST);
 
 	/* Enable copy and wait for completion */
 	regmap_writel(D0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL_EN_M, mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL);
